inc/api.class.php

```diff
@@ -283,22 +283,27 @@ protected function retrieveSession() {
     *   - 'entities_id': (default 'all') ID of the new active entity ("all" = load all possible entities). Optionnal
     *   - 'is_recursive': (default false) Also display sub entities of the active entity.  Optionnal
     *
-    * @return boolean
+    * @return array
     */
    protected function changeActiveEntities($params = []) {
 
       $this->initEndpoint();
 
       if (!isset($params['entities_id'])) {
-         $params['entities_id'] = 'all';
+         $entities_id = 'all';
+      } else {
+         $entities_id = intval($params['entities_id']);
       }
 
       if (!isset($params['is_recursive'])) {
          $params['is_recursive'] = false;
+      } else if (!is_bool($params['is_recursive'])) {
+         return $this->returnError();
       }
 
-      return Session::changeActiveEntities(intval($params['entities_id']),
-                                           $params['is_recursive']);
+      if (!Session::changeActiveEntities($entities_id, $params['is_recursive'])) {
```

Comment:
with this change, we have now no success response from the endpoint where we have ``true`` before.
Not sure if before answer was the best, but with this change, we have nothing.
Maybe, if the ``Session::changeActiveEntities(...)`` part return ok, we could call ``$this->getActiveEntities()``